platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java

```diff
@@ -449,13 +457,54 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
       // Scroll the map
       transform.moveBy(-distanceX, -distanceY, 0 /*no duration*/);
 
-      if (onScrollListener != null) {
-        onScrollListener.onScroll();
-      }
+      notifyOnScrollListeners();
       return true;
     }
   }
 
+  void notifyOnMapClickListeners(PointF tapPoint) {
+    // notify app of map click
+    if (onMapClickListener != null) {
+      onMapClickListener.onMapClick(projection.fromScreenLocation(tapPoint));
+    }
+
+    for (MapboxMap.OnMapClickListener listener : onMapClickListenerList) {
+      listener.onMapClick(projection.fromScreenLocation(tapPoint));
+    }
```

Comment:
With this setup we are still supporting the old system, could we add some inline comment to indicate to remove the top 3 lines when we remove support for it? It might confuse other people on the team looking at this code without having this context.

Reply:
An example of this in:
https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/CameraChangeDispatcher.java#L54